src/utils/validationUtils.ts

```diff
@@ -279,24 +279,99 @@ export const showFormErrors = <Values = unknown>({
   }
 };
 
-export const scrollToFirstError = ({
+export type ErrorFieldType =
+  | 'array'
+  | 'default'
+  | 'checkboxGroup'
+  | 'combobox'
+  | 'select'
+  | 'textEditor';
+
+type FieldList = Array<string | RegExp>;
+
+type FieldLists = {
+  arrayFields: FieldList;
+  checkboxGroupFields: FieldList;
+  comboboxFields: FieldList;
+  selectFields: FieldList;
+  textEditorFields: FieldList;
+};
+
+export type ErrorFieldAndType = {
+  fieldId: string;
+  type: ErrorFieldType;
+};
+
+const isInFieldList = (fieldList: FieldList, fieldName: string): boolean =>
+  Boolean(fieldList.find((field) => new RegExp(field).test(fieldName)));
+
+export const getFocusableFieldId = (
```

Comment:
This one could use a bit of refactoring to get rid of the `if-else if` structure (**code not tested**):
```
const fieldMappings: Record<keyof FieldLists, { suffix: string, type: ErrorFieldType }> = {
  comboboxFields: { suffix: '-input', type: 'combobox' },
  selectFields: { suffix: '-toggle-button', type: 'select' },
  textEditorFields: { suffix: '-text-editor', type: 'textEditor' },
  checkboxGroupFields: { suffix: '', type: 'checkboxGroup' },
  arrayFields: { suffix: '-error', type: 'array' },
};

export const getFocusableFieldId = (
  fieldName: string,
  fieldLists: FieldLists
): ErrorFieldAndType => {
  for (const [fieldListKey, { suffix, type }] of Object.entries(fieldMappings)) {
    if (isInFieldList(fieldLists[fieldListKey as keyof FieldLists], fieldName)) {
      return { fieldId: `${fieldName}${suffix}`, type };
    }
  }

  return { fieldId: fieldName, type: 'default' };
};
```

Reply:
Though this can be considered an optional change as this function has existed before with the `if-else if` structure.

Reply:
Really nice suggestion. Changed as suggested